### ISSUE TITLE
Categories are overriding each other:

### ISSUE THREAD

**Issue by Drachenfels** [isaacharrisholt/quiffen#23]
There is quite simple to reproduce bug with way you update categories, assuming qif file has this content:

```
!Type:Bank
D22/12/2017
T1000.00
PIncome For January
M
N
LIncome:Available next month
^
D01/01/2018
T1100.00
PStarting Balance
M
N
Cc
LIncome:Available this month
^
D02/01/2018
T-12.57
PFOOD C LONDON VIS
M
N
Cc
LEveryday Expenses:Food Budget
^
D02/01/2018
T-9.99
PGOUSTO.CO.UK - A LONDON  W3 VIS
M
N
Cc
LEveryday Expenses:Gousto
^
```

And script in the form:
```
#!/usr/bin/env python3

from quiffen import Qif

qif = Qif.parse("bug-example.qif")

print(qif.accounts)

acc = qif.accounts["Quiffen Default Account"]

for trs_type in acc.transactions.keys():
    print(f"Opening {trs_type}")

    transactions = acc.transactions[trs_type]
    print(f"Total number of transactions: {len(transactions)}")

for cat_name in qif.categories:
    tree = qif.categories[cat_name]
    print(tree.render_tree())

print("BYE!")
```

Output:
```
{'Quiffen Default Account': Account(name='Quiffen Default Account', desc='The default account created by Quiffen when no other accounts were present')}
Opening Bank
Total number of transactions: 3
Income (root)
└─ Available this month
Everyday Expenses (root)
└─ Gousto
BYE!
```

Expected Output:
```
{'Quiffen Default Account': Account(name='Quiffen Default Account', desc='The default account created by Quiffen when no other accounts were present')}
Opening Bank
Total number of transactions: 3
Income (root)
└─ Available this month
Everyday Expenses (root)
└─ Gousto
└─ Food Budget
BYE!
```

Bug is in the file: quiffen/core/qif.py, in following section:

```
            elif header_line.lower() in VALID_TRANSACTION_ACCOUNT_TYPES:
                # Other transaction type
                new_transaction, new_categories, new_classes = Transaction.from_string(
                    section,
                    separator=separator,
                    day_first=day_first,
                    line_number=line_number,
                )
                accounts[last_account].add_transaction(new_transaction, header_line)

                categories.update(new_categories)

                classes.update(new_classes)
```

If there is already parent category (Everyday Expenses) it will override it with a new one, hence always there will be one and only one child category.

**Comment by isaacharrisholt** [comment]
Thanks for bringing this to my attention. Unfortunately I don't have a huge amount of time to investigate this right now, but please feel free to submit a PR.

**Comment by isaacharrisholt** [comment]
Fixed in #36 as part of Quiffen v2
